src/main/java/abstracts/Car.java

```diff
@@ -0,0 +1,217 @@
+        System.out.println("Vehicle Turned On");
+    }
+
+    public void turnVehicleOff() {
+        System.out.println("Vehicle Turned Off");
+    }
+
+    public int changeToNextGear() {
+        if(actualGear <= numberOfGears){
+            actualGear++;
+        }else {
+            logger.warn("No more gears left to change");
+        }
+        return actualGear;
+    }
+
+    public int changeToPreviousGear() {
+        if(actualGear>=1){
+            actualGear--;
+        }else {
+            logger.warn("No more gears left to change");
+        }
+        return actualGear;
+    }
+
+    public void putOnNeutral() {
+        actualGear = 0;
+        System.out.println("Putting on neutral");
+    }
+
+
+    /**
+     * Class constructor
+     * @param fuel Fuel of car in liters
+     * @param numberOfGears Number of shift gears
+     * @param loadCapacity Trunk load capacity
+     * @param numberOfDoors Numeber of doors in the car
+     * @param maxNumberOfOccupants Maximum number of occupants
+     */
+    public Car(float fuel, int numberOfGears, float loadCapacity, int numberOfDoors, int maxNumberOfOccupants) {
+        this.fuel = fuel;
+        this.numberOfGears = numberOfGears;
+        this.loadCapacity = loadCapacity;
+        this.numberOfDoors = numberOfDoors;
+        this.maxNumberOfOccupants = maxNumberOfOccupants;
+    }
+
+    public float getFuel() {
+        return fuel;
+    }
+
+    public void setFuel(float fuel) {
+        this.fuel = fuel;
+    }
+
+    public int getNumberOfGears() {
+        return numberOfGears;
+    }
+
+    public void setNumberOfGears(int numberOfGears) {
+        this.numberOfGears = numberOfGears;
+    }
+
+    public int getActualGear() {
+        return actualGear;
+    }
+
+    public void setActualGear(int actualGear) {
+        this.actualGear = actualGear;
+    }
+
+    public float getLoadCapacity() {
+        return loadCapacity;
+    }
+
+    public void setLoadCapacity(float loadCapacity) {
+        this.loadCapacity = loadCapacity;
+    }
+
+    public int getNumberOfDoors() {
+        return numberOfDoors;
+    }
+
+    public void setNumberOfDoors(int numberOfDoors) {
+        this.numberOfDoors = numberOfDoors;
+    }
+
+    public int getMaxNumberOfOccupants() {
+        return maxNumberOfOccupants;
+    }
+
+    public void setMaxNumberOfOccupants(int maxNumberOfOccupants) {
+        this.maxNumberOfOccupants = maxNumberOfOccupants;
+    }
+
+
+    /**
+     *Check if the car is already full and try to add the occupant. If the occupant is already in the car it gets error.
+     *
+     * @param name name of occupant that will enter in the motorcycle
+     */
+    public void addOccupant(String name){
+
+        if (occupantsNames.size() < maxNumberOfOccupants){
+            try {
+                occupantsNames.add(name);
+            }catch (Exception e){
+                logger.error("Fail to insert new occupant. Already in car. " + e);
```

Comment:
When logging favor formatting parameters instead of string concat as in
`logger.error("Fail because of parameters {} {} {}", a, b, c, e);`

This saves a little time and memory when the logging utility is disable. Note that when logging error you dont need a specific '{}' for the exception as long as the exception is the last parameter.